types/chrome-remote-interface/index.d.ts

```diff
@@ -24,21 +57,89 @@ declare namespace CDP {
 
     interface Client {
         close: () => Promise<void>;
+        on(event: 'event', callback: (message: EventMessage) => void): void;
+        // '<domain>.<method>'
+        on<T extends keyof ProtocolMappingApi.Events>(event: T, callback: (params: ProtocolMappingApi.Events[T][0], sessionId?: string) => void): void;
+        // '<domain>.<method>.<sessionId>'
+        on(event: string, callback: (params: object, sessionId?: string) => void): void;
+        on(event: 'ready' | 'disconnect', callback: () => void): void;
```

Comment:
```suggestion
        on(event: 'ready' | 'disconnect', callback: () => void): void;
        // '<domain>.<method>'
        on<T extends keyof ProtocolMappingApi.Events>(event: T, callback: (params: ProtocolMappingApi.Events[T][0], sessionId?: string) => void): void;
        // '<domain>.<method>.<sessionId>'
        on(event: string, callback: (params: object, sessionId?: string) => void): void;
```

Overloads must always go from most specific to least specific. Here it probably doesn't matter because of the `on(event: string, ...` fallback but it's still a bug.

Reply:
https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#ordering